### PR TITLE
Window write support for SSD1673 

### PR DESCRIPTION
**Pull request by vanwinkeljan**
Added support for writing to a window instead of the complete display

### CODE REVIEW

**Comment by vanwinkeljan** [comment]
Addressed doc changes based on @dbkinder comments on PR #9149

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

**Comment by vanwinkeljan** [comment]
@jfischer-phytec-iot any input?

File: include/display.h [hunk 1]
```diff
@@ -48,6 +48,10 @@ enum display_screen_info {
 	 * Electrophoretic Display.
 	 */
 	SCREEN_INFO_EPD			= BIT(2),
+	/**
+	 * Screen has two alternating ram buffers
```
[the remark this answers is not in the document]
Comment (vanwinkeljan) [reply include/display.h]: No currently there is no ```display_flush``` function available in the display API.

But ```SCREEN_INFO_DOUBLE_BUFFER``` is jut an indication to the higher layers, e.g. incase only a partial update of the display buffer is performed the same partial update needs to happen on the second buffer to keep the display content consistent.

**Review by nashif — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: drivers/display/ssd1673.c [hunk 2]
```diff
@@ -415,49 +426,74 @@ static int ssd1673_controller_init(struct device *dev)
 	k_sleep(SSD1673_RESET_DELAY);
 	ssd1673_busy_wait(driver);
 
-	if (ssd1673_write_cmd(driver, SSD1673_CMD_SW_RESET, NULL, 0)) {
-		return -1;
+	err = ssd1673_write_cmd(driver, SSD1673_CMD_SW_RESET, NULL, 0);
+	if (err < 0) {
```
Comment (jfischer-phytec-iot) [inline drivers/display/ssd1673.c]: Please stop to add it, it does not really help and adds unnecessary code

File: drivers/display/ssd1673.c [hunk 3]
```diff
@@ -169,54 +173,32 @@ static int ssd1673_suspend(const struct device *dev)
 				 &tmp, sizeof(tmp));
 }
 
-static int ssd1673_update_display(const struct device *dev, bool initial)
```
Comment (jfischer-phytec-iot) [inline drivers/display/ssd1673.c]: Why are you revise this funktion? Why this change is necessary for region update?

File: drivers/display/ssd1673.c [hunk 4]
```diff
@@ -363,11 +320,8 @@ static int ssd1673_set_brightness(const struct device *dev,
 
 static int ssd1673_set_contrast(const struct device *dev, u8_t contrast)
```
Comment (jfischer-phytec-iot) [inline drivers/display/ssd1673.c]: It was a nice hack for initial update, why are you removing it? and why in this commit?

File: drivers/display/ssd1673.c [hunk 5]
```diff
@@ -225,118 +207,93 @@ static int ssd1673_write(const struct device *dev, const u16_t x,
...
-	sbuf.buf = (u8_t *)buf;
-	sbuf.len = desc->buf_size;
-	if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
-		return -1;
+	err = ssd1673_set_ram_ptr(driver, x_start, y_start);
+	if (err < 0) {
+		return err;
 	}
 
-	/* clear unusable page */
```
Comment (jfischer-phytec-iot) [inline drivers/display/ssd1673.c]: That must stay, you can not really use these lines.

**Review by jfischer-no — DISMISSED** [review]
Please split 171c4f440d392618e905eec78d4762ae457476c5 in logical peaces and remove unnecessary code changes, keep the driver design and structure.

File: drivers/display/ssd1673.c (hunk 5 above)
Comment (vanwinkeljan) [reply drivers/display/ssd1673.c]: There is no need to clear this lines on every write as it is done once on init

File: drivers/display/ssd1673.c (hunk 4 above)
Comment (vanwinkeljan) [reply drivers/display/ssd1673.c]: It was no longer needed as the initial clear is done at init

File: include/display.h [hunk 6]
```diff
@@ -48,6 +48,10 @@ enum display_screen_info {
...
+	 */
+	SCREEN_INFO_DOUBLE_BUFFER	= BIT(3),
```
[the remark this answers is not in the document]
Comment (vanwinkeljan) [reply include/display.h]: moved to #6826

File: drivers/display/ssd1673.c (hunk 3 above)
Comment (vanwinkeljan) [reply drivers/display/ssd1673.c]: parameter is no longer needed as init is done during display init

**Comment by vanwinkeljan** [comment]
@jfischer-phytec-iot Splitted commits as requested and provided replies where needed

**Comment by nashif** [comment]
@jfischer-phytec-iot feedback please

**Comment by vanwinkeljan** [comment]
@jfischer-phytec-iot any further input?

File: drivers/display/ssd1673.c (hunk 4 above)
Comment (jfischer-phytec-iot) [reply drivers/display/ssd1673.c]: This commit must be reverted because it removed possibility for the user to perform initial update.
